File: 2021/02-dive-2.js
const readline = require('readline')
const rl = readline.createInterface({ input: process.stdin })

let f=0
let d=0
let a=0

rl.on("line", data => {
  const [movement, X] = data.split(' ')
  const x = Number.parseInt(X)

  switch (movement) {
    case 'down':
      a = a + x
      break
    case 'up':
      a = a - x
      break
    case 'forward':
      f = f + x
      d = d + (a * x)
      break
  }
})

rl.on('close', () => {
  console.log(f*d)
})

File: 2021/02-dive-2.py
forward=0
depth=0
aim=0

with open('02-input.txt') as file:
    for line in file.readlines():
      [command, x] = line.split()
      x = int(x)
        
      if command == 'down':
          aim = aim + x
      elif command == 'up':
          aim = aim - x
      elif command == 'forward':
          forward = forward + x
          depth = depth + (aim * x)

print(forward*depth)
